cypress/support/validations/undefinedValidation.js

```diff
@@ -0,0 +1,90 @@
+            ? apiOrEventObject.eventName
+            : validationType == CONSTANTS.METHOD
+              ? apiOrEventObject.apiName
+              : null;
+        // Loop through each item of validations array of validation object
+        validationTypeObject.validations.forEach((validation) => {
+          // If the field to be validated is result, directly assert if result stored in response is undefined
+          if (validation.field === CONSTANTS.RESULT) {
+            fireLog.isUndefined(
+              methodOrEventResponse.result,
+              `Undefined Validation : Expected ${methodOrEventName} response to have the field ${validation.field} as undefined`
+            );
+          } else {
+            // Else recursilvely access the nested field properties and verify the corresponding value in response is undefined
+            const fieldParts = validation.field.split('.');
+            let currentObject = methodOrEventResponse;
+            // Throw error if any properties other than the provided field is undefined
+            for (let i = 0; i < fieldParts.length - 1; i++) {
+              const part = fieldParts[i];
+              if (!currentObject || typeof currentObject[part] === CONSTANTS.UNDEFINED) {
+                cy.log(
+                  `Undefined Validation : Expected ${methodOrEventName} response to have property ${part} in the field ${validation.field}`
+                ).then(() => {
+                  fireLog.assert(
+                    false,
+                    `Expected ${methodOrEventName} response to have property ${part} in the field ${validation.field}`
+                  );
+                });
+              }
+              currentObject = currentObject[part];
+            }
+            // Else assert if provided field is undefined
+            const finalPart = fieldParts[fieldParts.length - 1];
+            fireLog.isUndefined(
+              currentObject[finalPart],
+              `Undefined Validation : Expected ${methodOrEventName} response to have the field ${validation.field} as undefined`
+            );
+          }
+        });
+      } catch (error) {
+        cy.log(
+          `Undefined Validation : Received following error while performing validation of type undefined on response',
+        ${JSON.stringify(error)}`
+        ).then(() => {
+          fireLog.assert(
+            false,
+            `Undefined Validation : Received following error while performing validation of type undefined on response',
+          ${JSON.stringify(error)}`
+          );
+        });
+      }
+    } else {
+      cy.log(
+        `Undefined Validation : Expected validation object or api/event object stored in global list to not be undefined`
+      ).then(() => {
+        fireLog.assert(
+          false,
+          'Undefined Validation : Expected validation object or api/event object stored in global list to not be undefined'
+        );
+      });
+    }
```

Comment:
general comment on logging, cy.log and fireLog doesnt have to be used together. We can use fireLog to do print logs as well as assertions.